pkg/registry/extensions/daemonset/strategy_test.go

```diff
@@ -31,3 +37,71 @@ func TestDefaultGarbageCollectionPolicy(t *testing.T) {
 		t.Errorf("DefaultGarbageCollectionPolicy() = %#v, want %#v", got, want)
 	}
 }
+
+func TestSelectorImmutability(t *testing.T) {
+	tests := []struct {
+		method                 string
```

Comment:
Since this test is only for updates, the method should be `"PUT"` for all cases, so you don't need to specify it for each case.